articles/understanding-dart-loops/index.md

```diff
@@ -0,0 +1,331 @@
+
+
+***Example:***
+
+```Dart
+
+void main(){
+for(var i = 0; i<4; i++){
+    print ("Hello");
+ 
+ }
+
+}
+
+/** Output : Hello
+             Hello
+             Hello
+             Hello **/
+
+```
+Here the value of the initializer of the counter variable will increase by one from 0 based on the increment specified. If the value of i becomes equal to four or more, the loop terminates/ends since the condition states that the value of i should be less than 4. The loop is to print hello if the condition is true, hence the output is hello written four times (0,1,2,3) < 4. 
+
+2. #### While Loop
+
+While Loop is an example of [Indefinite Loops](#indefinite-loops).
+
+>Syntax of While Loop
+```Dart
+
+void Main(){
+   // initialize counter variable;
+
+   while(condition){
+      
+      //code;
+      
+      increment/decrement;
+
+   }
+
+}
+
+```
+
+>How the While Loop works.
+
+The flow of the While Loop: 
+
+**Condition Check - Code Execute - Increment**.
+
+In the While Loop, we initialization of the counter variable outside the syntax. It begins with the condition check, if it's true, the code executes. The value of the counter variable is then increments or decrements, and the new counter variable goes through the condition check. It will continue until the condition check is false.
+
+***NOTE:*** 
+#### We must always have the increment or decrement part of the While Loop, failure to which we create a never-ending loop.
+
+***Example:***
+
+```Dart
+
+void main(){
+  int i = 0;
+
+   while(i<3){
+      print("Hello");
+      i++;
+   }
+
+}
+
+/** Output: Hello
+            Hello
+            Hello **/
+
+```
+In the example above, our condition check begins with 0<3 which is true therefore the code executes and prints out the first hello. The value of i the increases by one to 1 and the condition 1<3 is still true, so the code executes. This goes on for one more loop (2<3) and ends when the condition is now (3<3) which is false. Hence three hellos get printed.
+
+3. #### Do-While Loop
+
+Do-While Loop is an example of [Indefinite Loops](#indefinite-loops).
+
+>Syntax of Do-While Loop.
+
+```Dart
+
+void main(){
+   initialize counter variable;
+   do{
+      //code;
+      increment/decrement;
+   }while(condition);
+
+}
+
+```
+
+>How the Do-While Loop works.
+
+The flow of the Do-While Loop: 
+
+**Code Execution - Increment - Condition Check**.
+
+In Do-While Loop the code executes then the counter variable initialized outside the syntax increments or decrements. Then the condition check, if it's met, it begins from code execution. The code execution will continue until the condition check turns false.
```

Comment:
You can also explicitly state that the code block inside do-while will definitely get executed at least once.

Reply:
Also, please break it down into points.